addons/survey/controllers/main.py

```diff
@@ -364,20 +374,30 @@ def survey_display_page(self, survey_token, answer_token, **post):
         return request.render('survey.survey_page_fill',
             self._prepare_survey_data(access_data['survey_sudo'], answer_sudo, **post))
 
-    @http.route('/survey/get_background_image/<string:survey_token>/<string:answer_token>', type='http', auth="public", website=True, sitemap=False)
-    def survey_get_background(self, survey_token, answer_token):
-        access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
-        if access_data['validity_code'] is not True:
-            return werkzeug.exceptions.Forbidden()
-
-        survey_sudo, answer_sudo = access_data['survey_sudo'], access_data['answer_sudo']
-
+    def _get_background_image(self, model_name, res_id):
         status, headers, image_base64 = request.env['ir.http'].sudo().binary_content(
-            model='survey.survey', id=survey_sudo.id, field='background_image',
+            model=model_name, id=res_id, field='background_image',
             default_mimetype='image/png')
-
         return request.env['ir.http']._content_image_get_response(status, headers, image_base64)
 
+    @http.route('/survey/<string:survey_token>/get_background_image',
+                type='http', auth="public", website=True, sitemap=False)
+    def survey_get_background(self, survey_token):
+        survey_sudo, dummy = self._fetch_from_access_token(survey_token, False)
+        return self._get_background_image(survey_sudo._name, survey_sudo.id)
+
+    @http.route('/survey/<string:survey_token>/<int:question_id>/get_background_image',
+                type='http', auth="public", website=True, sitemap=False)
+    def survey_question_get_background(self, survey_token, question_id):
+        survey_sudo, dummy = self._fetch_from_access_token(survey_token, False)
+
+        question = survey_sudo.question_and_page_ids.filtered(lambda q: q.id == question_id)
```

Comment:
Here we *always* have a section_id right?
Maybe rename to section_id and check only within the "page_ids" of the survey?

Reply:
indeed, this is a leftovers of previous implementation where the controller accepted question_ids